### DISCUSSION TITLE
Transaction retrry

### DISCUSSION THREAD

**Discussion by liningpan** [weiznich/diesel_async#170] [General]
The `async_bb8_diesel` crate developed by Oxide supports transaction retry, which is necessary 
when working with serializable transactions (especially cockroach db).

https://docs.rs/async-bb8-diesel/latest/async_bb8_diesel/trait.AsyncConnection.html

Would it possible to have something similar in `diesel_async`?

**Comment by weiznich** [comment]
I'm happy to accept a PR for this as long as it contains documentation + and a reasonable set of tests. Otherwise this is nothing I plan to work on in the next weeks/months for capacity reasons.
